inc/Validation/Validator.php

```diff
@@ -188,17 +188,27 @@ private function check_non_primitive_type( array $type, mixed $value ): bool|WP_
 					return $this->create_error( 'Class does not implement ArraySerializableInterface', $class_ref );
 				}
 
-				// The config can provide a `__subclass` property that indicates that we
-				// should inflate using a subclass of the specified class.
-				$subclass = $value['__subclass'] ?? null;
-				if ( null !== $subclass ) {
-					if ( ! is_subclass_of( $subclass, $class_ref, true ) ) {
-						return $this->create_error( 'Class specified by __subclass must be a subclass of the specified class', $subclass );
-					}
+				// The config must provide a `__class` property so that we know which
+				// class to inflate. This allows values to target subclasses of the
+				// specified class and also provides disambiguation when the type is
+				// used in a union type (one_of).
+				$subclass = $value[ ArraySerializableInterface::CLASS_REF_ATTRIBUTE ] ?? null;
+				if ( null === $subclass ) {
+					return $this->create_error( 'Value does not provide a __class property', $class_ref );
+				}
 
-					$class_ref = $subclass;
+				if ( ! class_exists( $subclass ) ) {
+					return $this->create_error( 'Class specified by __class property does not exist', $subclass );
 				}
 
+				if ( $subclass !== $class_ref && ! is_subclass_of( $subclass, $class_ref, true ) ) {
+					return $this->create_error( 'Class specified by __class property must match or be a subclass of the target class', $subclass );
```

Comment:
This would be a pretty abstract error to receive. Would it be possible to `sprintf()` what these classes should be? e.g.

```suggestion
					return $this->create_error( sprintf( 'Class specified by __class property "%s" must match or be a subclass of the target class %s', $subclass, $class_ref ), $subclass ) );
```

although I'm not sure if the string interpolation would work with these values.

Reply:
Also, something similar for the error above, too:

```
Class specified by __class property "%s" does not exist
```